s4e-web/src/app/utils/error-interceptor/error.interceptor.ts

```diff
@@ -2,23 +2,85 @@ import { Injectable } from '@angular/core';
 import {HttpRequest, HttpHandler, HttpEvent, HttpInterceptor, HttpErrorResponse} from '@angular/common/http';
 import { Observable, throwError } from 'rxjs';
 import { catchError } from 'rxjs/operators';
+import { Router } from '@angular/router';
+
+const ERROR_CODE_UNATHORIZED = 401;
+const ERROR_CODE_FORBIDEN = 403;
+const ERROR_CODE_BAD_REQUEST = 400;
+const ERROR_CODE_TIMEOUT = 101;
+
+export const ERROR_CODE_NOT_FOUND = 404;
+export const ERROR_CODE_INTERNAL_SERVER_ERROR = 500;
+export const ERROR_CODE_BAD_GATEWAY = 502;
+
+export const ERROR_INTERCEPTOR_SKIP_HEADER = 'Error-Interceptor-Skip-Header';
+
+
+class HttpErrorHelper {
+  public static isServerSideError = (error: HttpErrorResponse): boolean => error instanceof HttpErrorResponse;
+
+  public static isClientSideError = (error: HttpErrorResponse): boolean => error instanceof HttpErrorResponse
+    && error.error instanceof ErrorEvent
+
+  public static hasSkipHeader = (request: HttpRequest<any>): boolean => request.headers.has(ERROR_INTERCEPTOR_SKIP_HEADER);
+
+  public static skipedHttpEvent = (request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> => next
+    .handle(request.clone({ headers: request.headers.delete(ERROR_INTERCEPTOR_SKIP_HEADER) }))
+}
 
 @Injectable({
   providedIn: 'root',
 })
 export class ErrorInterceptor implements HttpInterceptor {
+  constructor(private _router: Router) {}
+
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-    return next.handle(request).pipe(catchError((error: HttpErrorResponse) => {
-      // switch(error.status) {
-      //   case 400: {
-      //     break;
-      //   }
-      //   default: {
-      //
-      //   }
-      // }
-      // const error = (error.error && error.error.message) || error.statusText;
-      return throwError(error);
-    }));
+    return next.handle(request)
+      .pipe(
+        catchError((error: HttpErrorResponse) => {
+          if (HttpErrorHelper.hasSkipHeader(request)) {
+            return HttpErrorHelper.skipedHttpEvent(request, next);
+          }
+
+          if (HttpErrorHelper.isClientSideError(error)) {
+            console.error('Error Event');
+            // TODO How to handle this error?
```

Comment:
We should pass it. I think it can only happen if there is another interceptor earlier in the chain, and it throws error.

Reply:
ok